src/Raven.Server/Documents/AI/AiGen/AbstractChatCompletionClient.cs

```diff
@@ -0,0 +1,458 @@
+
+        if (string.IsNullOrEmpty(content))
+        {
+            choice0.TryGet("finish_reason", out string finishReason);
+            choice0.TryGet("refusal", out string refusal);
+        
+            throw new GenAiRefusedToAnswerException("The request was refused by the model")
+            {
+                Refusal = refusal,
+                FinishReason = finishReason,
+                RequestId = GetRequestId(response.Headers)
+            };
+        }
+
+        if (responseContent.TryGet("usage", out BlittableJsonReaderObject usage) == false)
+            throw new GenAiUnexpectedResponseException("No choices property in response: " + responseContent)
+            {
+                RequestId = GetRequestId(response.Headers)
+            };
+
+        return (content, usage.ToString());
+    }
+
+    private HttpRequestMessage GetRequest(JsonOperationContext ctx, string prompt, string context)
+    {
+        var content = new BlittableJsonContent(async stream =>
+        {
+            await using (var writer = new AsyncBlittableJsonTextWriter(ctx, stream))
+            {
+                writer.WriteStartObject();
+
+                if (_forTestingPurposes?.ModifyPayload != null)
+                {
+                    _forTestingPurposes?.ModifyPayload.Invoke(writer);
+                    writer.WriteEndObject();
+                }
+
+                writer.WritePropertyName("model");
+                writer.WriteString(_model);
+                writer.WriteComma();
+        
+                writer.WritePropertyName("messages");
+                writer.WriteStartArray();
+                writer.WriteStartObject();
+                writer.WritePropertyName("role");
+                writer.WriteString("system");
+                writer.WriteComma();
+                writer.WritePropertyName("content");
+                writer.WriteString(prompt);
+                writer.WriteEndObject();
+                writer.WriteComma();
+                writer.WriteStartObject();
+                writer.WritePropertyName("role");
+                writer.WriteString("user");
+                writer.WriteComma();
+                writer.WritePropertyName("content");
+                writer.WriteString(context);
+                writer.WriteEndObject();
+                writer.WriteEndArray();
+                writer.WriteComma();
+        
+                writer.WritePropertyName("response_format");
+                writer.WriteStartObject();
+                writer.WritePropertyName("type");
+                writer.WriteString("json_schema");
+                writer.WriteComma();
+                writer.WritePropertyName("json_schema");
+                writer.WriteObject(await GetStructuredOutputSchemaAsBlittable());
+                writer.WriteEndObject();
+        
+                writer.WriteEndObject();
+            }
+        }, Conventions);
+
+        
+        content.Headers.Add("Content-Type", "application/json");
```

Comment:
use constant, also maybe better to set default header?

Reply:
Done